src/darsia/utils/linear_solvers/mg.py

```diff
@@ -0,0 +1,235 @@
+"""
+Multigrid solver for linear systems arising from discretizations of the
+minimization problem. Could be used as a preconditioner or a solver in itself.
+"""
+
+
+from typing import Optional, Union
+
+import numpy as np
+
+import darsia as da
+
+
+class MG(da.Solver):
+    """
+    Muiltilvel solver for the problem: mass_coeff * x - diffusion_coeff * laplace(x) = rhs
```

Comment:
Any restrictions on the size of the input array? How do you take care of sizes which are not a power of 2? Maybe a very short comment here could be added.

Reply:
There are no restrictions in princliple, but my method might not be very good.. Now I'm just disregarding the last row/column when this is the case..

Reply:
OK. Does in this case the prolongation add this row again, or does it always double?

Reply:
It actually always doubles, although it should have added the row again. I totally forgot about this. I will look at it again.

Reply:
Im just merging this with a note that it should be improved upon in the future.